filebeat/docs/multiline.asciidoc

```diff
@@ -0,0 +1,138 @@
+<<regexp-support>> to avoid common mistakes.
+
+==== Testing Your Regexp Pattern for Multiline
+
+To make it easier for you to test the regexp patterns in your multiline config, we've created a 
+http://play.golang.org/p/ABEJaX_lsK[Go Playground]. You can simply plug in the regexp pattern along with
+the `negate` setting that you plan to use, and paste a sample message between the content backticks (` `).
+Then click Run, and you'll see which lines in the message match your specified configuration. For example:
+
+image:images/go-playground.png[]
+
+==== Examples of Multiline Configuration
+
+The examples in this section cover the following use cases:
+
+* Combining a Java stack trace into a single event
+* Combining C-style line continuations into a single event
+* Combining multiple lines from time-stamped events
+
+===== Java Stack Traces
+
+Java stack traces consist of multiple lines, with each line after the initial line beginning with whitespace, as in
+this example:
+
+[source,java]
+-------------------------------------------------------------------------------------
+Exception in thread "main" java.lang.NullPointerException
+        at com.example.myproject.Book.getTitle(Book.java:16)
+        at com.example.myproject.Author.getBookTitles(Author.java:25)
+        at com.example.myproject.Bootstrap.main(Bootstrap.java:14)
+-------------------------------------------------------------------------------------
+
+To consolidate these lines into a single event in Filebeat, use the following multiline configuration:
+
+[source,json]
+-------------------------------------------------------------------------------------
+multiline:
+  pattern: '^[[:space:]]'
+  negate: false
+  match: after
+-------------------------------------------------------------------------------------
+
+This configuration merges any line that begins with whitespace up to the previous line.
+
+Here's a Java stack trace that presents a slightly more complex example: 
+
+["source","sh",subs="attributes,callouts"]
+-------------------------------------------------------------------------------------
+Exception in thread "main" java.lang.IllegalStateException: A book has a null property
+       at com.example.myproject.Author.getBookIds(Author.java:38)
+       at com.example.myproject.Bootstrap.main(Bootstrap.java:14)
+Caused by: java.lang.NullPointerException
+       at com.example.myproject.Book.getId(Book.java:22)
+       at com.example.myproject.Author.getBookIds(Author.java:35)
+       ... 1 more
+-------------------------------------------------------------------------------------
+
+To consolidate these lines into a single event in Filebeat, use the following multiline configuration:
+
+["source","sh",subs="attributes,callouts"]
+-------------------------------------------------------------------------------------
+multiline:
+    pattern: '^[[:space:]]+(at|...)|^Caused by:'
```

Comment:
this regex should become `^[[:space:]]+(at|\.\.\.)|^Caused by:` The `.` operator means, match any character. Due to requiring indentation in general, the pattern can be simplified to `^[[:space:]]+|^Caused by:` or `^[ \t]+|^Caused by:` .